app/src/main/java/com/markkolenbrander/capstonenewsapp/ArticleViewModel.kt

```diff
@@ -0,0 +1,39 @@
+package com.markkolenbrander.capstonenewsapp
+
+import androidx.lifecycle.*
+import com.markkolenbrander.capstonenewsapp.models.*
+import com.markkolenbrander.capstonenewsapp.networking.NewsService
+import com.markkolenbrander.capstonenewsapp.utils.CustomResult
+import kotlinx.coroutines.launch
+
+const val BASE_URL = "https://newsapi.org/v2/"
+const val API_TOKEN = "9ced23497a9d4184bffbe366d3a804d7"
+
+class ArticleViewModel(private val newsService: NewsService) : ViewModel() {
+
+    class Factory(
+        private val newsService: NewsService,
+    ): ViewModelProvider.Factory {
+        override fun <T : ViewModel> create(modelClass: Class<T>): T {
+            return ArticleViewModel(newsService) as T
+        }
+    }
+
+    private val _base = MutableLiveData<Base>()
+    val base: LiveData<Base> = _base
+
+
+    init {
+        viewModelScope.launch {
+            try {
+                val response = newsService.getArticles(API_TOKEN, Country.NL, Category.GENERAL)
+                _base.value = response
+
+            } catch (e: Exception){
+                CustomResult.Failure(e)
+            }
+        }
+
+    }
```

Comment:
nit-pick here but you could provide the viewModelScope's `launch` block with an `IO` dispatcher.  Retrofit automatically uses a different dispatcher to make the network API, so it's fine as-is but if you wanted to experiment with changing dispatchers you could do something like this:

```suggestion
    init {
        viewModelScope.launch(IO) {
            try {
                val response = newsService.getArticles(API_TOKEN, Country.NL, Category.GENERAL)
                withContext(Main) {
                    _base.value = response
                }

            } catch (e: Exception){
                CustomResult.Failure(e)
            }
        }

    }
```